feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c8ee68f280c0a58a3d439.md

---
id: 643c8ee68f280c0a58a3d439
title: Step 29
challengeType: 0
dashedName: step-29
---

# --description--

> To print the result on the page, target the HTML element with the `id` of `posts-container`.

Reviewer: ```suggestion
Use the `.getElementById()` method to get the `#posts-container` element. Assign it to a `postsContainer` variable.
```

This step doesn't actually print the result on the page. (:

Author: updated

# --hints--

> You should use a `const` variable named `postsContainer` to target the HTML element with the `id` of `posts-container`.
> 
> ```js
> assert(code.match(/const\s+postsContainer\s+=\s+document\.getElementById\(['"]posts-container['"]\)/g));
> ```

Reviewer: ````suggestion
You should use `const` to declare a `postsContainer` variable.

```js
assert.match(code, /const\s+postsContainer\s+=/);
```

You should use `document.getElementById()` to get the `#posts-container` element.

```js
assert.match(code, /document\.getElementById\(('|"|`)posts-container\1\)/);
```

You should assign the `#posts-container` element to your `postsContainer` variable.

```js
assert(code.match(/const\s+postsContainer\s+=\s+document\.getElementById\(('|"|`)posts-container\1\)/g));
```
````

Prefer granular steps. It helps campers identify where a typo might be, for example.

Author: fixed

---
id: 643c8ee68f280c0a58a3d439
title: Step 29
challengeType: 0
dashedName: step-29
---

# --description--

To print the result on the page, first you need to target the HTML element with the `id` of `posts-container`. like this 

```js

> const postsContainer = document.getElementById('posts-container');

Reviewer 2: Let's remove the example here too because campers will be comfortable with the `getElementById ` method by the time this project comes around.

Author: done

> ```

Author: done
